libraries/ESP8266WiFi/examples/BearSSL_ServerClientCert/BearSSL_ServerClientCert.ino

```diff
@@ -0,0 +1,259 @@
+xyMUZeiL3vyjqFnBruLPGiuDiHlrmUaeyFG7YaX5ht9rXg7Cciu//IKzAgMBAAEw
+DQYJKoZIhvcNAQELBQADggEBAEnG+FNyNCOkBvzHiUpHHpScxZqM2f+XDcewJgeS
+L6HkYEDIZZDNnd5gduSvkHpdJtWgsvJ7dJZL40w7Ba5sxpZHPIgKJGl9hzMkG+aA
+z5GMkjys9h2xpQZx9KL3q7G6A+C0bll7ODZlwBtY07CFMykT4Mp2oMRrQKRucMSV
+AB1mKujLAnMRKJ3NM89RQJH4GYiRps9y/HvM5lh7EIK/J0/nEZeJxY5hJngskPKb
+oPPdmkR97kaQnll4KNsC3owVlHVU2fMftgYkgQLzyeWgzcNa39AF3B6JlcOzNyQY
+seoK24dHmt6tWmn/sbxX7Aa6TL/4mVlFoOgcaTJyVaY/BrY=
+-----END CERTIFICATE-----
+)EOF";
+
+// Note there are no client certificates required here in the server.
+// That is because all clients will send a certificate that can be
+// proven to be signed by the public CA certificate included at the
+// head of the app.
+
+// Set time via NTP, as required for x.509 validation
+void setClock()
+{
+  configTime(3 * 3600, 0, "pool.ntp.org", "time.nist.gov");
+
+  Serial.print("Waiting for NTP time sync: ");
+  time_t now = time(nullptr);
+  while (now < 8 * 3600 * 2) {
+    delay(500);
+    Serial.print(".");
+    now = time(nullptr);
+  }
+  Serial.println("");
+  struct tm timeinfo;
+  gmtime_r(&now, &timeinfo);
+  Serial.print("Current time: ");
+  Serial.print(asctime(&timeinfo));
+}
+
+void setup() {
+  Serial.begin(115200);
+  Serial.println();
+  Serial.println();
+
+  // We start by connecting to a WiFi network
+  Serial.print("Connecting to ");
+  Serial.println(ssid);
+  WiFi.mode(WIFI_STA);
+  WiFi.begin(ssid, pass);
+
+  while (WiFi.status() != WL_CONNECTED) {
+    delay(500);
+    Serial.print(".");
+  }
+  Serial.println("");
+
+  Serial.println("WiFi connected");
+  Serial.println("IP address: ");
+  Serial.println(WiFi.localIP());
+
+  setClock(); // Required for X.509 validation
+
+  // Attach the server private cert/key combo
+  BearSSLX509List *serverCertList = new BearSSLX509List(server_cert);
+  BearSSLPrivateKey *serverPrivKey = new BearSSLPrivateKey(server_private_key);
+  server.setRSACert(serverCertList, serverPrivKey);
+
+  // Require a certificate validated by the trusted CA
+  BearSSLX509List *serverTrustedCA = new BearSSLX509List(ca_cert);
+  server.setClientTrustAnchor(serverTrustedCA);
+
+  // Actually start accepting connections
+  server.begin();
+}
+
+static const char *HTTP_RES =
+        "HTTP/1.0 200 OK\r\n"
+        "Connection: close\r\n"
+        "Content-Length: 59\r\n"
+        "Content-Type: text/html; charset=iso-8859-1\r\n"
+        "\r\n"
+        "<html>\r\n"
+        "<body>\r\n"
+        "<p>Hello my friend!</p>\r\n"
+        "</body>\r\n"
+        "</html>\r\n";
+
+void loop() {
+  BearSSL::WiFiClientSecure incoming = server.available();
+  if (!incoming) {
+    return;
+  }
+  Serial.println("Incoming connection...\n");
+  
+  // Ugly way to wait for \r\n (i.e. end of HTTP request which we don't actually parse here)
+  uint32_t timeout=millis() + 1000;
+  int lcwn = 0;
+  for (;;) {
+    unsigned char x=0;
+    if (millis() > timeout) {
+      goto client_drop;
```

Comment:
Remove goto

Reply:
You and `GOTO` must have had a serious falling out in 8-bit `BASIC` days, @devyte! I will rewrite these instances.  Does the SDK support `setjmp()`?  :)

Reply:
I normally use goto's for bailout paths in C code, but I agree that in this specific case (example sketch) some different approach could be used.

setjmp is supported, yes, just make sure setjmp and longjmp are called on the same stack (i.e. not from cont to sys or vice versa).

Reply:
I was just kidding about `setjmp`!  This just needs a different flow structure, not a poor-man's exception handler. :)

Reply:
> serious falling out in 8-bit BASIC days
>] Something like that, but in recent C/C++ world. 
I do think that there is a place for goto in programming. However, my firm position is that it is a dangerous tool, meant for responsible adults, and should be strictly kept away from children, lest you end up faced with code from hell. We're the responsible adults here.
What I mean by all this is that, if there is ever a goto to be added into our core, the place would have to be in code that is very far away from the user, and certainly not in an example sketch meant for the Arduino user audience.  And even then, it should be well justified, and with big fat comments explaining the reasoning and code logic behind it.

About my falling out, here are two of my horror stories:
- In my previous job, I came across a goto from the method of one class to a label inside a method of an entirely different class. The code actually worked, until you changed any one line of code in those two methods, then it would mysteriously and randomly crash.
- In my current job, I was asked to partially reverse engineer a 20 year old geometric algorithm implemented as a sort of state machine of 200+ states, where each state was a block of code. Changing states was done with if(some condition) then goto some label, where the labels were named something like "goto1", "goto2", "goto3"..... Zero function use, zero comments across the whole thing, and zero documentation.